aws/logs_monitoring/lambda_function.py

```diff
@@ -277,8 +283,20 @@ def s3_handler(event, context, metadata):
             )
             yield structured_line
     else:
+        # Check if using multiline log regex pattern
+        if DD_MULTILINE_LOG_REGEX_PATTERN:
+            split_data = multiline_regex.split(data)
+
+            # handle case where lambda is passed both line and pattern separated logs
+            # if there is only a single log and that log does not start with multiline regex pattern
+            # assume that these are line separated logs, not pattern separated
+            if len(split_data) <= 1 and not multiline_regex_start_pattern.match(data):
```

Comment:
I would probably remove this check and change this check https://github.com/DataDog/datadog-serverless-functions/pull/111/files#diff-618731ddf1c446ef45d09613cb4b189bR287 to:
```
if DD_MULTILINE_LOG_REGEX_PATTERN and multiline_regex_start_pattern.match(data):
```

Reply:
also I think that we should change a bit `multiline_regex_start_pattern` to make sure it starts from the beginning i.e.:
```
multiline_regex = re.compile("^{}".format(DD_MULTILINE_LOG_REGEX_PATTERN))
```

Reply:
Thanks, updated this to clean up the nested if statements. Fwiw, I think [`.match`](https://docs.python.org/3/library/re.html#re.match) implicitly ensures the pattern starts from the beginning but i agree it's good to explicitly put this in the regular expression, updated to reflect that.